dockstore-webservice/src/main/java/io/dockstore/webservice/helpers/GitHubSourceCodeRepo.java

```diff
@@ -1354,6 +1354,21 @@ public void updateReferenceType(String repositoryId, Version version) {
         }
     }
 
+    public String getHash(String repositoryId, String reference) {
+        try {
+            GHRepository repo = github.getRepository(repositoryId);
+            GHRef ref = repo.getRef(reference);
+            return ref.getObject().getSha();
+        } catch (GHFileNotFoundException e) {
+            LOG.info("Could not find reference '{}' on repository '{}'", reference, repositoryId);
+            return null;
+        } catch (IOException e) {
+            LOG.error(gitUsername + ": IOException on getHash " + e.getMessage(), e);
+            throw new CustomWebApplicationException("Could not access GitHub reference", HttpStatus.SC_INTERNAL_SERVER_ERROR);
```

Comment:
This doesn't seem to match line 1388. 
In other words, a deleted commit would seem to be ignored, a deleted tag would seem to throw a 500.

Reply:
Follow-up: Does the catch on line https://github.com/dockstore/dockstore/pull/5809/files#diff-f85a33452ebb0f60ef1029aabf9e19310f028e512444d47d11a94c0581d7676dL538 mean that the `CustomWebApplicationException` would be caught and the non-accessible reference would be processed as a push anyway?

Reply:
> Follow-up: Does the catch on line https://github.com/dockstore/dockstore/pull/5809/files#diff-f85a33452ebb0f60ef1029aabf9e19310f028e512444d47d11a94c0581d7676dL538 mean that the `CustomWebApplicationException` would be caught and the non-accessible reference would be processed as a push anyway?

Yes, if we have a problem determining whether or not we should process a push or delete, we process it, like we did before the ref inspection code.

Reply:
> This doesn't seem to match line 1388. In other words, a deleted commit would seem to be ignored, a deleted tag would seem to throw a 500.

This method is derived from the `getCommitID` method at line 1399.  When we try to look up the tag, and it's not there, the GitHub code should throw a `GHFileNotFoundException`, and this method will return `null`.  The next `catch` will throw a `CustomWebApplicationException` on other errors like network problems.  We need the two catches to be able to gracefully differentiate the two types of conditions.

Reply:
Ah I see, I didn't realize the method was overloaded